Userland/Libraries/LibJS/Runtime/VM.h

```diff
@@ -144,10 +147,9 @@ class VM : public RefCounted<VM> {
         return index < arguments.size() ? arguments[index] : js_undefined();
     }
 
-    Value this_value(Object& global_object) const
+    Value this_value() const
     {
-        if (m_execution_context_stack.is_empty())
-            return &global_object;
+        VERIFY(!m_execution_context_stack.is_empty());
         return running_execution_context().this_value;
     }
```

Comment:
Our this_value seems a little non-spec like anyways, but that's not something that should be fixed here.

Reply:
The weird thing is that the spec doesn't link the text anywhere when saying "the this value", but given it's styled like this:

![image](https://user-images.githubusercontent.com/19366641/186129542-5da161ca-7783-4a37-aaf2-3d8f39adefbe.png)

I suppose they refer to this:

![image](https://user-images.githubusercontent.com/19366641/186129650-8cd6407c-2071-40c6-af58-30682f547be7.png)

https://tc39.es/ecma262/#sec-this-keyword-runtime-semantics-evaluation

So I suppose we really should be getting it from the environment's `GetThisBinding`, instead of having an extra field for it. Either way, since `GetThisEnvironment` relies on the running execution context, assuming we have one here should be fine.

Reply:
Yeah the `VERIFY(...)` is fine, I just looked at the spec to check it and found we weren't really following it here, more a comment for future-someone-who-will-fix-this-value (me?).